test/e2e/vmagent_test.go

```diff
@@ -291,3 +302,411 @@ func createExternalNodeCRD(data *TestData, nodeName string, ifName string, ip st
+			t.Errorf("Failed to run iperf asR client, retrying: %v", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	return err
+}
+
+func runIperfServer(t *testing.T, data *TestData, vm vmInfo, dstPort int32) error {
+	cmd := getIperf3Command(vm.osType, nil, dstPort, false, true)
+	cmdStr := strings.Join(cmd, " ")
+	if vm.osType == windowsOS {
+		cmdStr = fmt.Sprintf(`cmd.exe /c "%s"`, cmdStr)
+	}
+	_, _, _, err := data.provider.RunCommandOnNode(vm.nodeName, cmdStr)
+	if err != nil {
+		t.Errorf("Failed to run iperf3 command on VM %s, err %v", vm.nodeName, err)
+		return err
+	}
+	t.Logf("Run iperf3 server on VM %s with command %s", vm.nodeName, cmdStr)
+	return nil
+}
+
+func stopIperfCommand(t *testing.T, data *TestData, vm vmInfo) error {
+	cmdStr := `cmd.exe /c "taskkill /IM iperf3.exe /F"`
+	if vm.osType == linuxOS {
+		cmdStr = "pkill iperf3"
+	}
+	_, _, _, err := data.provider.RunCommandOnNode(vm.nodeName, cmdStr)
+	if err != nil {
+		t.Errorf("Failed to stop iperf3 command on VM %s, err %v", vm.nodeName, err)
+		return err
+	}
+	t.Logf("Stoped iperf3 on VM %s", vm.nodeName)
+	return nil
+}
+
+func runIperfClient(t *testing.T, data *TestData, targetVM vmInfo, svrIP net.IP, dstPort int32, isUDP bool, connected bool, ruleAction crdv1alpha1.RuleAction) error {
+	cmd := getIperf3Command(targetVM.osType, svrIP, dstPort, isUDP, false)
+	cmdStr := strings.Join(cmd, " ")
+	if targetVM.osType == windowsOS {
+		cmdStr = fmt.Sprintf(`cmd.exe /c "%s"`, cmdStr)
+	}
+	expectedOutput := "iperf Done"
+	if !connected {
+		switch ruleAction {
+		case crdv1alpha1.RuleActionDrop:
+			expectedOutput = "Connection timed out"
+		case crdv1alpha1.RuleActionReject:
+			if isUDP {
+				expectedOutput = "No route to host"
+			} else {
+				expectedOutput = "Connection refused"
+			}
+		}
+	}
+
+	errCh := make(chan error, 0)
+	go func() {
+		err := runCommandAndCheckResult(data, targetVM.nodeName, cmdStr, expectedOutput, "")
+		if err != nil {
+			t.Logf("Failed to run iperf3 client command on VM %s: %v", targetVM.nodeName, err)
+		}
+		errCh <- err
+	}()
+
+	select {
+	// Complete the iperf3 command in 10s forcibly if it does not return. The stuck in iperf3 command possibly happens
+	// when it is running as a client on Windows, if the port opened on iperf server is blocking by ANP rule. To avoid
+	// the test is blocking, we forcibly stop the client. As the iperf client is configured with parameter "-t 2" meaning
+	// the utility is expected to send packets in 2s. The force quit should not break the testing workloads.
+	case <-time.After(time.Second * 10):
+		t.Logf("Iperf3 command %s does not return in 10s, stop it forcibly", cmdStr)
+		err := stopIperfCommand(t, data, targetVM)
+		assert.Nil(t, err, fmt.Sprintf("Failed to stop iperf3 command after 10s: %v", err))
+		if !connected {
+			return nil
+		}
+		return fmt.Errorf("failed to complete iperf3 command %s in 10s", cmdStr)
```

Comment:
is it simpler to specify a connection timeout by `--connect-timeout` (note the unit is millisecond) for iperf3 and wait for result synchronously?